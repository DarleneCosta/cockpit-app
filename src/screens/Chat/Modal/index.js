import React, { Component } from "react";
import { Text, TouchableOpacity, StyleSheet, View } from "react-native";
import Modal from "react-native-modal";

const ModalConfirmation = () => {
  const [visibleModal, setVisibleModal] = useState(false);

  const renderButton = (text, onPress) => (
    <TouchableOpacity onPress={onPress}>
      <View style={styles.button}>
        <Text>{text}</Text>
      </View>
    </TouchableOpacity>
  );

  const renderModalContent = () => (
    <View style={styles.modalContent}>
      <Text>Hello!</Text>
      {renderButton("Close", () => this.setState({ visibleModal: null }))}
    </View>
  );

  return (
    <View style={styles.container}>
      {renderButton("Bottom half modal", () => setVisibleModal(true))}
      <Modal isVisible={visibleModal} style={styles.bottomModal}>
        {renderModalContent()}
      </Modal>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  button: {
    backgroundColor: "lightblue",
    padding: 12,
    margin: 16,
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 4,
    borderColor: "rgba(0, 0, 0, 0.1)",
  },
  modalContent: {
    backgroundColor: "white",
    padding: 22,
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 4,
    borderColor: "rgba(0, 0, 0, 0.1)",
  },
  bottomModal: {
    justifyContent: "flex-end",
    margin: 0,
  },
});
