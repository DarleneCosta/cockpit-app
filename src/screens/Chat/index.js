import { ScrollView, Text, View } from "react-native";
import { useRoute } from "@react-navigation/native";
import BackButton from "./../../components/BackButton";
import Title from "./../../components/Title";
import Identification from "./../../components/Identification";
import styles from "./style";
import { MaterialIcons } from "@expo/vector-icons";
import { GiftedChat, Send, Bubble } from "react-native-gifted-chat";
import { useState, useCallback, useEffect } from "react";

const Chat = () => {
  const route = useRoute();
  const { complainId } = route.params;

  const [messages, setMessages] = useState([]);

  useEffect(() => {
    setMessages([
      {
        _id: 1,
        text: "OlÃ¡! Em que posso ser Ãºtil?",
        createdAt: new Date(),
        user: {
          _id: 2,
          name: "Atendente",
          avatar:
            "https://ps.w.org/simple-user-avatar/assets/icon-256x256.png?rev=2413146",
        },
        isTyping: true,
      },
    ]);
  }, []);

  const onSend = useCallback((messages = []) => {
    setMessages((previousMessages) =>
      GiftedChat.append(previousMessages, messages)
    );
  }, []);

  const renderSend = (props) => {
    return (
      <Send {...props} containerStyle={styles.send}>
        <MaterialIcons name="send" size={30} color="white" />
      </Send>
    );
  };

  const renderBubble = (props) => {
    return <Bubble {...props} containerStyle={styles.message} />;
  };

  return (
    <>
      <BackButton />
      <Title text={"Chat com Atendente"} />
      <View style={styles.container}>
        <Identification label="Protocolo" valueID={complainId} />
      </View>

      <GiftedChat
        containerStyle={styles.chat}
        placeholder={"Envie sua mensagem aqui..."}
        messages={messages}
        onSend={(messages) => onSend(messages)}
        renderBubble={renderBubble}
        renderSend={renderSend}
        user={{
          _id: 1,
        }}
      />
    </>
  );
};

export default Chat;
