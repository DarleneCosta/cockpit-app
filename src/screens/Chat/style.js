import { StyleSheet } from "react-native";
import colors from "./../../styles/colors";

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 30,
  },
  chat: {
    flex: 1,
  },
});

export default styles;
