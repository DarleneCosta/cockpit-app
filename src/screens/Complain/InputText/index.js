import { useEffect, useState } from "react";
import { Text, TextInput } from "react-native";
import styles from "./style";

const InputText = ({ value, onChangeText, maxLength=30,  msgRequired='', autoFocus=false, multiline=false, placeholder="" }) => {
    const [showMsg, setShowMsg] = useState(false)    
    const [focus, setFocus] = useState(false)
    
    const handleRequired = () =>{
        setFocus(false)
       const required = msgRequired && !value
       setShowMsg(required)
    }

    const handleOnFocus =() =>{ 
        setFocus(true)    
        setShowMsg(false)
      }

  return (
    <>
        <TextInput
            style={styles(showMsg, focus).input}
            onChangeText={onChangeText}
            value={value}
            maxLength={maxLength}
            multiline={multiline}
            placeholder={placeholder}
            autoFocus={autoFocus}
            onBlur={handleRequired}
            onFocus = {handleOnFocus}
        />
        {showMsg && <Text  style={styles().labelError}>{msgRequired}</Text>}
  </>

  );
};

export default InputText;
