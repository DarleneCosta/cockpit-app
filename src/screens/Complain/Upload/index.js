import { Text, TouchableOpacity, View } from "react-native";
import styles from "./style";
import * as ImagePicker from 'expo-image-picker';

const Upload = () => {
  return (
    <View style={styles.container}>
      <TouchableOpacity>
        <View style={styles.button}>
          <Text style={styles.labelButton}>Enviar Imagem</Text>
        </View>
      </TouchableOpacity>
      <Text style={styles.label}>Nenhuma Imagem Selecionada</Text>
    </View>
  );
};

export default Upload;
