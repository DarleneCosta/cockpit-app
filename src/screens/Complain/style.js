import { StyleSheet } from "react-native";
import colors from "./../../styles/colors";

const styles = StyleSheet.create({
  container: {
    marginHorizontal: 30,
  },
  input: {
    borderColor: colors.primary.main,
    borderWidth: 1,
    borderRadius: 25,
    paddingHorizontal: 18,
    fontSize: 16,
    width: "100%",
    paddingVertical: 14,
    marginBottom: 10,
  },
  inputText: {
    borderColor: colors.primary.main,
    borderWidth: 1,
    borderRadius: 25,
    paddingHorizontal: 18,
    fontSize: 16,
    width: "100%",
    paddingVertical: 14,
    marginBottom: 10,
  },
  label: {
    fontSize: 15,
    fontWeight: "bold",
    marginVertical: 10,
  },
  button: {
    marginBottom: 5,
  },
});

export default styles;
