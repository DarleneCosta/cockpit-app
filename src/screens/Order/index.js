import { View, Text, ScrollView, TouchableOpacity } from "react-native";
import { useNavigation } from "@react-navigation/native";
import styles from "./style";
import { Ionicons } from "@expo/vector-icons";
import Divider from "../../components/Divider";
import useOrder from "../../hooks/useOrder";
import Summary from "./Summary";
import colors from "../../styles/colors";
import Timeline from "./Timeline";
import Title from "../../components/Title";
import Details from "./Details";
import Payments from "./Payments";
import Adresses from "./Adresses";
import ButtonTemplate from "./../../components/ButtonTemplate";
import ButtonWhats from "./../../components/ButtonWhats/index";

const Order = () => {
  const order = useOrder();
  const navigation = useNavigation();

  const handlePressButton = () => {
    navigation.navigate("SolicitarAtendimento", { orderId: order.id });
  };

  return (
    order && (
      <>
        <ScrollView>
          <TouchableOpacity style={styles.wrapperMenu}>
            <Text style={styles.menu}>MENU</Text>
            <Ionicons name="chevron-down-outline" size={25} color="black" />
          </TouchableOpacity>

          <Title text={`COMPRA #${order.id}`} />

          <View style={styles.card}>
            <Summary order={order} />
            <Divider color={colors.primary.main} margin={24} />
            <Timeline historic={order.historic} />
            <Divider color={colors.primary.main} margin={24} />
            <Details items={order.items} />
            <Divider color={colors.primary.main} margin={24} />
            <Payments
              payments={order.payments}
              delivery={order.delivery}
              amount={order.value}
            />
            <Divider color={colors.primary.main} margin={10} />
            <Adresses adresses={order.adresses} />
          </View>
          <View style={styles.wrapperFooter}>
            <Text style={styles.labelFooter}>
              Algum problema com sua compra?
            </Text>
          </View>
          <TouchableOpacity onPress={handlePressButton}>
            <ButtonTemplate label={"Fale Conosco"} />
          </TouchableOpacity>
        </ScrollView>
        <ButtonWhats />
      </>
    )
  );
};

export default Order;
