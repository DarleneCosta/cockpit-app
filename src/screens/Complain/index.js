import { useEffect, useState } from "react";
import {
  View,
  ScrollView,
  Text,
  TouchableOpacity,
  LayoutAnimation,
} from "react-native";
import { useRoute, useNavigation } from "@react-navigation/native";

import Title from "./../../components/Title";

import styles from "./style";
import ButtonBack from "../../components/ButtonBack";
import ButtonTemplate from "./../../components/ButtonTemplate";
import ButtonWhats from "./../../components/ButtonWhats";
import Identification from "./../../components/Identification";
import Upload from "./Upload";
import Spinner from "react-native-loading-spinner-overlay";
import ComboBox from "./ComboBox";
import InputText from "./InputText";
import { useTypesComplain } from "../../hooks/useComplain";
import utils from "../../utils";
//import { createdComplain } from "../../services/repositories/complain";

const Complain = () => {
  LayoutAnimation.linear();

  const typesComplain = useTypesComplain();
  const navigation = useNavigation();
  const route = useRoute();

  const { orderId } = route.params;
  const [titleComplain, setTitleComplain] = useState("");
  const [describeComplain, setDescribeComplain] = useState("");
  const [image, setImage] = useState(null);
  const [type, setType] = useState("");
  const [spinner, setSpinner] = useState(false);
  const [disabled, setDisabled] = useState(false);

  useEffect(() => {
    handleValid();
  }, [titleComplain, type]);

  const handleValid = () => {
    const isValid = !!titleComplain && !!type;
    setDisabled(!isValid);
    return isValid;
  };

  const handlePressButton = async () => {
    const isValid = handleValid();
    if (!isValid) {
      return;
    }

    try {
      setSpinner(true);
      // const { id: complainId } = await createdComplain({
      //   orderId,
      //   titleComplain,
      //   describeComplain,
      //   type,
      //   image
      // })
      navigation.navigate("Atendimento", { complainId:utils.random()});
    } finally {
      setSpinner(false);
    }
  };

  return (
    <>
      <Spinner
        visible={spinner}
        textContent={"Aguarde..."}
        textStyle={styles.spinnerText}
      />
      <ScrollView>
        <ButtonBack />
        <Title text="Fale Conosco" />

        <View style={styles.container}>
          <Identification label="Id do pedido" valueID={orderId} />

          <Text style={styles.label}>Motivo do contato *</Text>
          <InputText
            autoFocus={true}
            onChangeText={setTitleComplain}
            value={titleComplain}
            maxLength={30}
            msgRequired="Por favor preencha"
            placeholder="Digite o motivo do contato"
          />

          <Text style={styles.label}>Tipo *</Text>
          <ComboBox
            data={typesComplain}
            setValue={setType}
            value={type}
            force={disabled}
            msgRequired="Por favor informe qual a categoria se enquandra"
          />

          <Text style={styles.label}>Descrição</Text>
          <InputText
            placeholder="Detalhe o motivo do contato"
            onChangeText={setDescribeComplain}
            value={describeComplain}
            maxLength={1000}
            multiline={true}
          />

          <Text style={styles.label}>Adicione evidências aqui</Text>
          <Upload image={image} setImage={setImage} />
        </View>
        <TouchableOpacity        
          style={styles.button}
          onPress={handlePressButton}
          disabled={disabled}
        >
          <ButtonTemplate label={"Enviar Mensagem"} disabled={disabled} />
        </TouchableOpacity>
      </ScrollView>
      <ButtonWhats orderId={orderId} />
    </>
  );
};

export default Complain;
