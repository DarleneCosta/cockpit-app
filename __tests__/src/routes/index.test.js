
import { View } from 'react-native';
import { render } from '@testing-library/react-native';

import Routers from '../../../src/routers';
import Order from '../../../src/screens/Order';
import Complain from '../../../src/screens/Complain';
import Chat from '../../../src/screens/Chat';

jest.mock('../../../src/screens/Order');
jest.mock('../../../src/screens/Complain');
jest.mock('../../../src/screens/Chat');

describe('Routers', () => {
  it('should render correctly', () => {

    Order.mockImplementation(() => <View testID="Order" />);
    Complain.mockImplementation(() => <View testID="Complain" />);
    Chat.mockImplementation(() => <View testID="Chat" />);

    const { toJSON } = render(<Routers />);

    expect(toJSON()).toMatchSnapshot();
  });
});

