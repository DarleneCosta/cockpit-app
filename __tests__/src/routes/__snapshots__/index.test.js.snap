// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routers should render correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Compra"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            testID="Order"
          />
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
